test/DeploymentHelper.test.ts

```diff
@@ -0,0 +1,222 @@
+    ).connect(deployer);
+    deploymentParams = {
+      foreignToken,
+      merkleRoot,
+      multiTokenMediatorHome: multiTokenMediatorHome.address,
+      gnoToken: gnoToken.address,
+      wrappedNativeToken: wrappedNativeToken.address,
+      communityFundsTarget,
+      gnoPrice,
+      nativeTokenPrice,
+    };
+  });
+
+  describe("constructor parameters", async function () {
+    beforeEach(async function () {
+      await multiTokenMediatorHome.mock.bridgedTokenAddress.returns(
+        bridgedCowTokenAddress,
+      );
+      deploymentHelper = await DeploymentHelper.deploy(
+        ...constructorInput(ContractName.DeploymentHelper, deploymentParams),
+      );
+    });
+
+    it("has expected merkle root", async () => {
+      expect(await deploymentHelper.merkleRoot()).to.equal(merkleRoot);
+    });
+
+    it("has expected multiTokenMediator", async () => {
+      expect(await deploymentHelper.multiTokenMediator()).to.equal(
+        multiTokenMediatorHome.address,
+      );
+    });
+
+    it("has expected real deploymentHelper", async () => {
+      expect(await deploymentHelper.foreignToken()).to.equal(foreignToken);
+    });
+
+    it("sets the expected community funds target", async function () {
+      expect(await deploymentHelper.communityFundsTarget()).to.equal(
+        communityFundsTarget,
+      );
+    });
+
+    it("sets the expected GNO deploymentHelper", async function () {
+      expect(await deploymentHelper.gnoToken()).to.equal(gnoToken.address);
+    });
+
+    it("sets the expected GNO price", async function () {
+      expect(await deploymentHelper.gnoPrice()).to.equal(gnoPrice);
+    });
+
+    it("sets the expected wrapped native deploymentHelper", async function () {
+      expect(await deploymentHelper.wrappedNativeToken()).to.equal(
+        wrappedNativeToken.address,
+      );
+    });
+
+    it("sets the expected native deploymentHelper price", async function () {
+      expect(await deploymentHelper.nativeTokenPrice()).to.equal(
+        nativeTokenPrice,
+      );
+    });
+  });
+
+  describe("deploy", async function () {
+    it("reverts if the bridge contract does not yet exists", async () => {
+      await multiTokenMediatorHome.mock.bridgedTokenAddress.returns(
+        constants.AddressZero,
+      );
+      deploymentHelper = await DeploymentHelper.deploy(
+        ...constructorInput(ContractName.DeploymentHelper, deploymentParams),
+      );
+      await expect(deploymentHelper.deploy()).to.be.revertedWith(
+        "cowToken not yet bridged",
+      );
+    });
+
+    it("deploys a new contracts if the bridge contract does exists", async () => {
+      await multiTokenMediatorHome.mock.bridgedTokenAddress.returns(
+        bridgedCowTokenAddress,
+      );
+      deploymentHelper = await DeploymentHelper.deploy(
+        ...constructorInput(ContractName.DeploymentHelper, deploymentParams),
+      );
+      await expect(deploymentHelper.deploy()).to.be.not.reverted;
+    });
```

Comment:
It would also be nice to test that the returned address is the address of the deployed token.

Reply:
I think we are doing this in all the follow-up unit tests.